tensorflow_probability/python/experimental/mcmc/sample_fold.py

```diff
@@ -20,9 +20,8 @@
 
 # Dependency imports
 import tensorflow.compat.v2 as tf
-from tensorflow_probability.python.experimental.mcmc import with_reductions
-from tensorflow_probability.python.experimental.mcmc import SampleDiscardingKernel
-from tensorflow_probability.python.experimental.mcmc import step_kernel
+from tensorflow_probability.python.experimental import mcmc
+from tensorflow_probability.python.experimental.mcmc.with_reductions import WithReductionsKernelResults
```

Comment:
Why not just add WithReductionsKernelResults to the experimental.mcmc `__init__` file?